app/presenters/webauthn_setup_presenter.rb

```diff
@@ -82,11 +83,19 @@ def nickname_label
     end
   end
 
+  def device_nickname
+    if @platform_authenticator
+      recent_devices.first.nice_name
```

Comment:
Is it possible to pass in the device name from the constructor? I think it'd be preferable since the device name from the user agent would be more direct compared to querying the database (and also saves a database query).

Is there concern that the existing unique requirement on name could create issues if it can't be overridden?

Reply:
Thanks for that suggestion. I'm still reworking how best to handle that but I'm no longer hitting recent_devices.

There is a method in WebauthnSetupForm that checks if the name is unique. I can try to handle appending to a duplicate name there.

Reply:
I think it will need to be handled, yeah